dot/digest/digest.go

```diff
@@ -267,23 +265,29 @@ func (h *Handler) persistBABEDigestsForNextEpoch(finalizedHeader *types.Header)
 	}
 
 	nextEpoch := currEpoch + 1
-	err = h.epochState.FinalizeBABENextEpochData(nextEpoch)
-	if err != nil && !errors.Is(err, state.ErrEpochNotInMemory) {
-		return fmt.Errorf("cannot finalize babe next epoch data for block number %d (%s): %w",
-			finalizedHeader.Number, finalizedHeader.Hash(), err)
+
+	appliedEpochData, appliedConfigData, err := h.epochState.AlreadyDefined(nextEpoch)
+	if err != nil {
+		return fmt.Errorf("cannot check if next epoch is already defined: %w", err)
 	}
```

Comment:
I'm kind of confused still as to why this is needed - a block won't be finalized twice, so what is the case where the digest data would already be set in the database? also, it would be cleaner to handle the digest check + writing to disk all the in `EpochState` instead of here.